React & Redux group project - Space Travelers' Hub

denisdiaconu opened the pull request:
We build an application that allows users to book rockets and join selected space missions.

We completed all the requirements: 

-    Use React documentation.
-    Use React components.
-    Use React props.
-    Use React Router.
-    Connect React and Redux.
-    Handle events in a React app.
-    Write unit tests with React Testing Library.
-    Use styles in a React app.
-    Use React life cycle methods.
-    Apply React best practices and language style guides in code.
-    Use store, actions and reducers in React.
-    Perform a code review for a team member.

denisdiaconu committed:
Initial commit

denisdiaconu committed:
Add api functions and reducer

denisdiaconu committed:
Add components

denisdiaconu committed:
Fix axios problem

denisdiaconu committed:
Add rockets and missions actions

denisdiaconu committed:
Finish all pages logic

CollinsTatang committed:
Set up configStore

CollinsTatang committed:
Add rockets handler and action

denisdiaconu committed:
Finish the logic behind add and remove

denisdiaconu committed:
Fix small error inside myprofile

CollinsTatang committed:
update reducer action

CollinsTatang committed:
Add rocket useEffect components

CollinsTatang committed:
Add checkBotton

CollinsTatang committed:
Add display rocket

CollinsTatang committed:
Add getRockets

CollinsTatang committed:
Merge pull request #26 from CollinsTatang/feature-rocket

Feature rocket

CollinsTatang committed:
Merge branch 'dev' into feature

CollinsTatang committed:
Merge pull request #27 from CollinsTatang/feature

Feature

CollinsTatang committed:
Add getMissions api url and method

CollinsTatang committed:
install react-bootstrap

CollinsTatang committed:
Style the header nav

denisdiaconu committed:
Add tests with jest

denisdiaconu committed:
Add fix for second describe

denisdiaconu committed:
Add display for reserved in rockets

denisdiaconu committed:
Add display for rockets status

CollinsTatang committed:
Add missions style

CollinsTatang committed:
add rocket style

CollinsTatang committed:
update name

denisdiaconu committed:
Add style for my profile

denisdiaconu committed:
Merge pull request #28 from CollinsTatang/feature-styling

Feature styling

denisdiaconu committed:
Merge branch 'dev' into testing

denisdiaconu committed:
Merge pull request #29 from CollinsTatang/testing

Testing

denisdiaconu committed:
Fix linters errors

denisdiaconu committed:
Update README.md

denisdiaconu committed:
Update README.md

denisdiaconu committed:
Fix stylelint errors

MahmoudBakr23 commented on src/api/api.js:
```diff
@@ -0,0 +1,44 @@
+const missionsUrl = 'https://api.spacexdata.com/v3/missions';
+
+const getRockets = async () => await axios.get(`${rocketsUrl}`).then((result) => {
+  const rockets = [];
```
- [x] **OPTIONAL**
I see that you defined a constant rockets variable that its value is an empty array to store the fetched data from the API endpoint 👏 But isn't the data itself represent an array of objects! 😄 For example: if you `console.log(result.data[0]);`, you should see the first rocket object in your browser console. This would make less code 🙏 

_Please consider this point with the missions as well._ 🙏

MahmoudBakr23 commented on src/api/api.js:
```diff
@@ -0,0 +1,44 @@
+        id: rocketData.rocket_id,
+        name: rocketData.rocket_name,
+        description: rocketData.description,
+        flickr_images: rocketData.flickr_images,
```
- [x] **OPTIONAL**
In fetching the flicker images part, I noticed that you're saving the whole images array in the Redux store, then select the first image when rendering 👏 How about receiving the first image of the array directly and it'll be the only image stored in the Redux store! 😄 This could be done by `flicker_image: rocketData.flicker_images[0],` Please tell me what you think? 🙏

MahmoudBakr23 commented on README.md:
```diff
@@ -1 +1,77 @@
+# Space Travelers' Hub
+
+In this task, we worked with the real live data from the SpaceX API. Our task was to build a web application for a company that provides commercial and scientific space travel services. The application will allow users to book rockets and join selected space missions.
+
```
- [x] Isn't going to be better if you attached an image or two of your great app in the README file to be marked as a 100% README file? 😉 💯

MahmoudBakr23 commented on src/tests/missions.react.test.js:
```diff
@@ -0,0 +1,68 @@
+  removeReservationRocket,
+} from '../redux/api/api';
+
+describe('Testing actions of receiving data', () => {
```
- [x] Well done testing the APIs and actions from your app's functionalities 👏 However, please add more unit testing and checking using React Testing Library and Jest. 😄 🙏 

_For example tests for rendering your components, checking elements, and inside functions._

MahmoudBakr23 commented on src/components/Missions.js:
```diff
@@ -0,0 +1,80 @@
+  function checkButton(status) {
+    if (status.reserved === false) {
+      return (
+        <Button variant="outline-danger" onClick={() => dispatch(addReservationMission(status))}>Join Mission</Button>
```
- [x] Nicely done using Bootstrap for styling 😉 However, you might replace the styling between this button and the other one. According to the given design, the `Join Mission` button is `secondary`, and the other one is `danger` 😄 🙏 
_You can check the image out from [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=The%20Missions%20section%20displays%20a%20list%20of%20current%20missions%20along%20with%20their%20brief%20description%20and%20participation%20status.%20There%20is%20also%20a%20button%20next%20to%20each%20mission%20that%20allows%20users%20to%20join%20the%20selected%20mission%20or%20leave%20the%20mission%20the%20user%20joined%20earlier.)._

MahmoudBakr23 commented on src/components/Rockets.js:
```diff
@@ -0,0 +1,73 @@
+              <h1>{element.name}</h1>
+              <div className="specialRocket">
+                {checkStatus(element)}
+                <p className="desc">{element.description}</p>
```
- [x] Would you please double-check on the styling of this paragraph on the 'Rockets' page, because it's slightly different from the original design, there's a non-pleasant gap to the eyes that'd be much nicer if it's gone 😄 🙏 
![Screenshot](https://user-images.githubusercontent.com/47625407/132037962-08acfae5-f21e-4c57-aae1-389a2012566f.png)

_You can check the image right [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=The%20Rockets%20section%20displays%20a%20list%20of%20all%20available%20SpaceX%20rockets.%20Users%20can%20book%20each%20rocket%20by%20clicking%20the%20reservation%20button%20or%20cancel%20the%20previously%20made%20booking.%20The%20same%20layout%20is%20used%20to%20form%20the%20Dragons%20section%20%5Badded%20only%20if%20your%20team%20has%203%20members%5D.)._

MahmoudBakr23 reviewed (changes requested):
Hello team😃, 

Great work and effort you put into this project so far. 🔥 

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct
- Descriptive PR headline, summary, and commit messages
- Workflow followed
- Great code separation (logic is only in reducers)
- Good handling with the APIs
- Well done using Redux middleware (thunk and logger)
- Well done creating actions and reducers

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

### Video Presentation Feedback

- [x] Great job covering the features of your application. However, I highly suggest that you both record another short video that highlights something that you guys are proud of or got you interested in developing this app. 😄 🙏 This could be fun to think about together 😉 

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Please check my **OPTIONAL** comments down below._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

denisdiaconu committed:
Improved the rockets and missions page

denisdiaconu committed:
Update README.md

denisdiaconu commented:
Thank you for the review!
We fixed the following problems:

-  In fetching the flicker images part.
- We attached two images.
- We added 8 tests for testing rendering.
- We changed the styling of the two buttons.
- The paragraph of the rockets page is inline now.

mricanho reviewed (approved):
# Status: ✔️ Project Approved ✔️ 

Hello  @denisdiaconu @CollinsTatang 🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

![awesome-job](https://user-images.githubusercontent.com/60631456/132057829-abcb22a8-4901-4b76-9e9a-2e38f9fe309e.gif)

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
